internal/services/managedredis/managed_redis_data_source.go

```diff
@@ -145,6 +146,24 @@ func (r ManagedRedisDataSource) Attributes() map[string]*pluginsdk.Schema {
 						},
 					},
 
+					"persistence": {
+						Type:     pluginsdk.TypeList,
+						Computed: true,
+						Elem: &pluginsdk.Resource{
+							Schema: map[string]*pluginsdk.Schema{
+								"method": {
+									Type:     pluginsdk.TypeString,
+									Computed: true,
+								},
+
+								"backup_frequency": {
+									Type:     pluginsdk.TypeString,
+									Computed: true,
+								},
```

Comment:
Maybe we shall consider the following schema:

```
"aof": {
  Optional: true,
  ExactlyOneOf: ...,
  {
    "backup_frequency": {},
  },
},

"rdb": {
  Optional: true,
  ExactlyOneOf: ...,
  {
    "backup_frequency": {},
  },
}
```

In this way, we can get rid of the customize diff and additional enum definitions in the validate package.

Reply:
I'm not sure about that design. If you look at the portal UX, this is designed as a single-choice radio button. In Terraform a block is normally used to group a set of properties.

It should be the swagger / API design that's changed to optimise client use, but I guess it's harder at this stage.